### PR TITLE
Add rowAuxClick to DatagridRow component

### PR DESCRIPTION
**Pull request by WiXSL**

### CODE REVIEW

File: packages/ra-ui-materialui/src/list/datagrid/DatagridRow.tsx [hunk 1]
```diff
@@ -128,6 +129,43 @@ const DatagridRow: FC<DatagridRowProps> = React.forwardRef((props, ref) => {
...
+            if (!rowAuxClick) return;
+            event.preventDefault();
+
+            const effect =
+                typeof rowAuxClick === 'function'
+                    ? await rowAuxClick(id, basePath, record)
+                    : rowAuxClick;
+            switch (effect) {
+                case 'edit':
+                    history.push(linkToRecord(basePath, id));
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/datagrid/DatagridRow.tsx:143]: the middle click is supposed to open the link in a new tab. I don't see that in your PR.

**Review by fzaninotto — CHANGES_REQUESTED** [review]

**Comment by fzaninotto** [comment]
Also, this is a new feature, and it should b PRed against `next` instead of `master`

File: packages/ra-ui-materialui/src/list/datagrid/DatagridRow.tsx [hunk 2]
```diff
@@ -150,6 +188,7 @@ const DatagridRow: FC<DatagridRowProps> = React.forwardRef((props, ref) => {
                 style={style}
                 hover={hover}
                 onClick={handleClick}
+                onAuxClick={handleAuxClick}
```
Comment (djhi) [inline packages/ra-ui-materialui/src/list/datagrid/DatagridRow.tsx:191]: I don't think this event handle the CTRL + click combo. But this combo might not be universal cross OSs. Any thoughts?

Comment (WiXSL) [reply packages/ra-ui-materialui/src/list/datagrid/DatagridRow.tsx:169]: Doesn't work in Safari
https://caniuse.com/?search=auxclick

Comment (djhi) [reply packages/ra-ui-materialui/src/list/datagrid/DatagridRow.tsx:169]: The new IE :joy:

Comment (WiXSL) [reply packages/ra-ui-materialui/src/list/datagrid/DatagridRow.tsx:162]: Hehe. What should I do? Is this ok, or I should change the implementation?

Comment (fzaninotto) [reply packages/ra-ui-materialui/src/list/datagrid/DatagridRow.tsx:162]: why don't you just modify the `DatagridRow::handleClick` method to detect if the ctrlKey is pressed, and switch from history.push() to window.open in that case?

Comment (WiXSL) [reply packages/ra-ui-materialui/src/list/datagrid/DatagridRow.tsx:162]: Ok

Comment (WiXSL) [reply packages/ra-ui-materialui/src/list/datagrid/DatagridRow.tsx]: I can't detect the button clicked using `onClick` handler, i have to use `onMosueDown`

**Comment by lefuturiste** [comment]
@fzaninotto The PR was again closed automatically (I assume). Can you reopen it?

**Pull request closed without merging**